File: app/src/main/java/com/example/carlauncher/data/local/LabRecord.java
package com.example.carlauncher.data.local;


import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(
        tableName = "lab_record",
        indices = {@Index(value = {"name"}, unique = true)}
)
public class LabRecord {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    public long id;

    @NonNull
    public String name = "";

    @NonNull
    public String value = "";

    @ColumnInfo(name = "created_at")
    public long createdAt;

    @NonNull
    @ColumnInfo(defaultValue = "''")
    public String note = "";
}

File: app/src/main/java/com/example/carlauncher/data/local/LabRecord.kt
package com.example.carlauncher.data.local

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "lab_record", indices = [Index(value = ["name"], unique = true)])
open class LabRecord {
    @field:PrimaryKey(autoGenerate = true)
    @field:ColumnInfo(name = "_id")
    @JvmField
    var id: Long = 0

    @JvmField var name: String = ""

    @JvmField var value: String = ""

    @field:ColumnInfo(name = "created_at") @JvmField var createdAt: Long = 0

    @field:ColumnInfo(defaultValue = "''") @JvmField var note: String = ""
}
